// react-sort-component/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactSortUtils = require('./react-sort-utils');

var ReactSortUtils = _interopRequireWildcard(_reactSortUtils);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ReactSort = function ReactSort(_ref) {
    var dataSource = _ref.dataSource,
        children = _ref.children,
        sortOptions = _ref.sortOptions,
        renderHeader = _ref.renderHeader;

    var sortedDataSource = ReactSortUtils.sortDataSource(sortOptions, dataSource);
    return _react2.default.createElement(
        'div',
        null,
        renderHeader && typeof renderHeader == 'function' ? renderHeader(ReactSortUtils, sortOptions, sortedDataSource) : null,
        sortedDataSource.map(function (item) {
            return children(item);
        })
    );
};

ReactSort.propTypes = {
    dataSource: _react.PropTypes.array.isRequired,
    sortOptions: _react.PropTypes.shape({
        sortField: _react.PropTypes.string.isRequired,
        sortDir: _react.PropTypes.oneOf(['asc', 'des']).isRequired,
        isCaseSensitive: _react.PropTypes.boolean
    }).isRequired,
    renderHeader: _react.PropTypes.func,
    renderFooter: _react.PropTypes.func
};

exports.default = ReactSort;
